src/lib/server/subscriptionUtils.ts

```diff
@@ -24,12 +24,17 @@ export const getStripeCustomerWithSubscriptions = async (uid: string) => {
     if (stripeCustomerId === null)
         return stripeCustomerId;
     return await stripe.customers.retrieve(stripeCustomerId, {
-        expand: ['subscriptions']
+        expand: ['subscriptions'],
     });
 }
 export const getBlendProSubscription = (customer: Stripe.Customer) => customer.subscriptions?.data.find((subscription) => subscription.items.data.find(({plan: { product, active }}) => active && product === STRIPE_BLEND_PRO_PRODUCT_CODE));
 
 export const getCustomerPortalSession = (customer: Stripe.Customer, returnUrl: string) => stripe.billingPortal.sessions.create({
     customer: customer.id,
     return_url: returnUrl
-})
\ No newline at end of file
+});
+
+export const hasCustomerSubscribedBefore = async (customer: Stripe.Customer) =>
+  (await stripe.subscriptions.list({ customer: customer.id, status: 'all', expand: ['data.items'] })).data.some(
+    (subscription) => subscription.items.data.some((item) => item.price.product === STRIPE_BLEND_PRO_PRODUCT_CODE),
+  );
```

Comment:
I guess there are multiple ways to get to the subscriptions list, but I think we should be internally consistent and do it in the same way as in the `getBlendProSubscription` method above. As an added benefit, if the `customer` object already has an array of subscriptions (which we make sure of by fetching the customer via `getStripeCustomerWithSubscriptions` above), we don't need another fetch call so this should be synchronous and faster. 

And if we are using the same matching function in two places we should split it out:
```suggestion
const isBlendProSubscription = (subscription: Stripe.Subscription) => subscription.items.data.some(({plan: { product }}) => product === STRIPE_BLEND_PRO_PRODUCT_CODE)

export const getBlendProSubscription = (customer: Stripe.Customer) => customer.subscriptions?.data.filter(({ plan: {active} }) => active).find(isBlendProSubscription)

export const hasCustomerSubscribedBefore = (customer: Stripe.Customer) => customer.subscriptions?.data.some(isBlendProSubscription)
  );
```

Reply:
makes sense, okay let me make this change!